app/src/main/java/com/example/triptracker/view/map/MapPopups.kt

```diff
@@ -136,3 +156,177 @@ fun AddressText(mpv: MapPopupViewModel, latitude: Float, longitude: Float) {
+          Row(modifier = Modifier.fillMaxWidth()) {
+            // change the image to the user's profile picture
+            AsyncImage(
+                model = userofpost.profileImageUrl,
+                contentDescription = "User Avatar",
+                modifier =
+                    Modifier.size(avatarSize)
+                        .clip(CircleShape)
+                        .testTag("ProfilePic")
+                        .clickable { /* TODO bring user to profile page */})
+
+            Spacer(modifier = Modifier.width(15.dp))
+            Text(
+                text = userofpost.name,
+                fontFamily = FontFamily(Font(R.font.montserrat_regular)),
+                //                wrapContentHeight(align = Alignment.CenterVertically),
+                fontWeight = FontWeight.Normal,
+                fontSize = 18.sp,
+                color = md_theme_grey,
+                modifier =
+                    Modifier.testTag("Username")
+                        .wrapContentHeight(align = Alignment.CenterVertically))
+            Spacer(Modifier.weight(1f))
+
+            Icon(
+                imageVector = Icons.Outlined.Star,
+                contentDescription = "Star",
+                Modifier.size(40.dp))
+          }
+          Spacer(modifier = Modifier.height(20.dp))
+          Text(
+              text = itinerary.title,
+              fontFamily = FontFamily(Font(R.font.montserrat_regular)),
+              fontWeight = FontWeight.Bold,
+              fontSize = 30.sp,
+              color = md_theme_light_onPrimary,
+              modifier = Modifier.testTag("Title"))
+          Text(
+              text = "${itinerary.flameCount}🔥",
+              color = md_theme_orange, // This is the orange color
+              fontFamily = FontFamily(Font(R.font.montserrat_regular)),
+              fontSize = 20.sp,
+              modifier =
+                  Modifier.padding(
+                      bottom = 20.dp,
+                      top = 10.dp,
+                  ))
+          Column(
+              modifier =
+                  Modifier.padding(top = 10.dp, start = 10.dp, end = 10.dp, bottom = 50.dp)) {
+                for (pin in itinerary.pinnedPlaces) {
+                  Text(text = "• ${pin.name}", color = md_theme_grey, fontSize = 20.sp)
+                  Spacer(modifier = Modifier.height(3.dp))
+                }
+              }
+
+          LazyRow {
+            items(itinerary.pinnedPlaces) { pin ->
+              for (image in pin.image_url) {
+                AsyncImage(
+                    model = image,
+                    contentDescription = pin.description,
+                    modifier = Modifier.clip(RoundedCornerShape(corner = CornerSize(12.dp))))
+
+                Spacer(modifier = Modifier.width(15.dp))
+              }
+            }
+          }
+        }
+
+        Button(
+            onClick = { /* Do something! */},
```

Comment:
what does this button then ? why it isn't link to the screen already implemented after ?

Reply:
that's not covered in this pr this is simply the ui